### DISCUSSION TITLE
Drift detected: Your database schema is not in sync with your migration history - nothing changed ...

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#24886] [Q&A]
### Question

I have these 2 migrations:

![image](https://github.com/user-attachments/assets/6ba00932-5e87-4fa9-860d-048af5b2cc7c)

1) This one applied without issue on the 18th,
2) This one is failing to apply with the message:
![image](https://github.com/user-attachments/assets/f9ce385d-3381-42f0-8c58-e91e9292173c)

The migration for the second one is this:
```sql
-- AlterTable
ALTER TABLE `marketplace` ADD COLUMN `loginMethod` INTEGER NOT NULL DEFAULT 0;
```
And the prisma change for this migration:
![image](https://github.com/user-attachments/assets/70f857b9-7136-4756-935c-3ef263d0ac01)

The tables it's complaining about are these:
![image](https://github.com/user-attachments/assets/bc47518d-820a-41b3-ab8a-27c89d948fc6)

If I run the following SQL prior to migration:
```sql
DROP INDEX marketplace_companyId_fkey ON marketplace;
DROP INDEX global_filter_companyId_fkey ON global_filter;
``` 
it'll work and the migration will apply.

This is a cut down version of each of the models in the schema:

```prisma
model Marketplace {
  id          Int      @id @default(autoincrement())
  flag        Int
  active      Boolean

  company   Company @relation(fields: [companyId], references: [id])
  companyId Int

  @@index([companyId, flag])
  @@index(active)
  @@map("marketplace")
}

model GlobalFilter {
  id        Int     @id @default(autoincrement())
  name      String
  favourite Boolean

  company   Company? @relation(fields: [companyId], references: [id])
  companyId Int?

  @@index([companyId, name])
  @@index([companyId, favourite])
  @@map("global_filter")
}
```

Prisma version info:
![image](https://github.com/user-attachments/assets/5c8dc0d2-18af-4ff8-86c5-4753e094e458)

If I create a new DB from scratch and apply all migrations, it doesn't add the companyId indexes to those 2 tables so I understand why it's complaining but I don't understand why now and why it doesn't add the indexes but does for other tables setup the same.

Why is this suddenly starting to complain about these indexes?

### How to reproduce (optional)

I don't know ... It's just started happening.

### Expected behavior (optional)

For the migration to apply without complaining about the indexes

### Information about Prisma Schema, Client Queries and Environment (optional)

I'd prefer not to post my schema. I've provided snippets for the bits that count.

**Comment by ludralph** [comment]
Hi @webnoob 👋 

Thank you for raising this question.

This issue may arise due to the interaction between the existing indexes and the new migration that adds the `loginMethod` column with a default value. When adding a new column with a `NOT NULL` constraint and a default value, the database needs to update all existing rows to set this default value. This can sometimes cause issues if there are existing constraints or indexes that may be indirectly affected.

You may also want to check if there is a diff by running the `migrate diff` command.

```
npx prisma migrate diff --from-schema-datamodel ./schema.prisma --to-schema-datasource ./schema.prisma --script
```
If there are truly no changes, then did you by chance run `prisma db push` instead of `prisma migrate dev`. Running `db push` and later `migrate dev` has been shown to cause a schema drift.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
